<?xml version="1.0"?>
<launch>
  <arg name="script" default="nav_cloning_node.py"/>
  <arg name="mode" default="use_dl_output"/>
  <arg name="map_file" default="willowgarage.yaml"/>
  <arg name="waypoints_file" default="willow_loop.yaml"/>
  <arg name="dist_err" default="0.8"/>
  <arg name="initial_pose_x" default="-10.78" />
  <arg name="initial_pose_y" default="-16.78" />
  <arg name="initial_pose_a" default="0.0" />
  <arg name="use_waypoint_nav" default="false" />
  <arg name="use_initpose" default="false" />
  <arg name="robot_x" default="0.0" />
  <arg name="robot_y" default="0.0" />
  <arg name="robot_Y" default="0.0" />
  <rosparam param="/nav_cloning_node/mode" subst_value="True">$(arg mode)</rosparam>


  <group if="$(arg use_initpose)">
    <node pkg="nav_cloning" type="initpose" name="initpose"/>
  </group>

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="5 nav_cloning nav_cloning.launch script:=$(arg script)"
        name="timed_roslaunch" output="screen" />

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="5 orne_navigation_executor play_waypoints_nav_gamma.launch 
        map_file:=$(find nav_cloning)/maps/$(arg map_file) 
        waypoints_file:=$(find nav_cloning)/maps/$(arg waypoints_file) 
        dist_err:=$(arg dist_err) 
        initial_pose_x:=$(arg initial_pose_x) 
        initial_pose_y:=$(arg initial_pose_y) 
        initial_pose_a:=$(arg initial_pose_a) 
        use_waypoint_nav:=$(arg use_waypoint_nav)"
        name="timed_roslaunch2" >
  </node>

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="20 nav_cloning start_wp_nav.launch"
        name="timed_roslaunch3" >
  </node>
</launch>
